[react] Add comment about removed index signature

frenic opened the pull request:
The [index signature for `React.CSSProperties` was removed](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/24911) to enable closed typing. In this PR I just added a comment about that and how to use module augmentation to add CSS custom properties or a index signature of their own.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

frenic commented:
cc @pelotom

pelotom commented on types/react/index.d.ts:
```diff
@@ -897,7 +897,27 @@ declare namespace React {
+        /**
+         * The index signature was removed to enable closed typing for style
+         * using CSSType. You're able to use module augmentation to add other
+         * properties, like CSS Custom Properties, or add a index signature
```
“...or add an* index signature...”

frenic replied:
Thanks 👍

pelotom commented on types/react/index.d.ts:
````diff
@@ -897,7 +897,27 @@ declare namespace React {
+         *     [index: string]: any;
+         *   }
+         * }
+         * ```
````
Maybe also good to mention that a cheap alternative, esp. for missing properties where you don't need to reference it in multiple places, is casting:

```ts
<div style={{ ['overflowAnchor' as any]: 'auto' }}>
  ...
</div>
```

frenic replied:
I added that to the README a  `csstype`. Thanks 👍

pelotom commented:
How about adding a comment to the effect of "If you find that these typings are missing a valid CSS property or value, please check whether an issue has already been filed for it at https://github.com/frenic/csstype/issues and if not, create one."

frenic commented:
@pelotom I changed the comment a bit and added a [link to the README of `csstype`](https://github.com/frenic/csstype#what-should-i-to-do-when-i-get-type-errors) where they can find exampes and more information. It's easier to maintain that README rather than this comment.

johnnyreilly commented:
Thanks!

AdamTReineke commented:
This helped me debug a compile error today after updating React, thanks you!
